database/soldiers-repository.js

```diff
@@ -0,0 +1,43 @@
+import { client } from '../connections.js';
+
+export const dbName = 'CallOfDuty';
+export const soldiersDBCollection = 'Soldiers';
+
+export async function addNewSoldier(newSoldier) {
+  const soldierToInsert = {
+    _id: newSoldier.id,
+    rank: newSoldier.rank,
+    name: newSoldier.name,
+    limitations: newSoldier.limitations,
+    duties: [],
+  };
+  const result = await client.db(dbName).collection(soldiersDBCollection)
+    .insertOne(soldierToInsert);
+  return result;
+}
+
+export async function lookForSoldier(specificSoldier) {
```

Comment:
i think you have a mistake. did all the test pass?
fix the solider route. you dont need to use the lookForSolider in the post route

Reply:
fixed